app/services/katello/repository_type_manager.rb

```diff
@@ -0,0 +1,59 @@
+module Katello
+  class RepositoryTypeManager
```

Comment:
oh yeah, so in the foreman plugin mode, you're doing:

Plugin.register()

and then Plugin.registered_plugins  to fetch them.  There is no PluginManager. So i think it makes sense in this caes to just have  RepositoryType.register